File: linearSearch.java
import java.util.Random;
import java.util.Scanner;
import java.util.Arrays;

public class linearSearch{
    public static int[] linear_search(int[] array, int sn){
        int checks = 0;
        for (int i = 0; i < array.length; i++){
            checks ++;
            if (array[i] == sn)
                return new int[]{i, checks};
        }
        return new int[] {-1, checks};
    }
    public static void main(String[] args){
        Scanner input = new Scanner(System.in);
        Random random = new Random();

        System.out.print("Repeat test how many times: ");
        int rep = input.nextInt();
        input.close();

        for (int i=0; i<rep; i++){
            int[] arr = new int[50];
            for (int j=0; j<50; j++)
                arr[j] = random.nextInt(50) + 1;

            System.out.println(Arrays.toString(arr));

            int sn = random.nextInt(50) + 1;
            int[] result = linear_search(arr, sn);
            int index = result[0];
            int checks = result[1];

            System.out.printf("Search number: %d | Linear search: %d elements checked\n", sn, checks);
            System.out.println();
        }
    }
}

File: linearSearch.py
import random
import time

def linearSearch(arr, sn):
    s = 0
    for i in range(len(arr)):
        s += 1
        if arr[i] == sn:
            return i, s
    return -1, s

arr = [random.randint(1, 25) for n in range(25)]
print(arr)
sn = random.randint(1, 25)
index , lChecks = linearSearch(arr, sn)
print(f"Search number: {sn} \nIndex: {index} \nLinear search: {lChecks} elements checked\n")